pkg/controller/framework/shared_informer.go

```diff
@@ -34,35 +34,28 @@ import (
 // have your item.  This has advantages over the broadcaster since it allows us
 // to share a common cache across many controllers. Extending the broadcaster
 // would have required us keep duplicate caches for each watch.
-type SharedInformer interface {
+type SharedIndexInformer interface {
 	// events to a single handler are delivered sequentially, but there is no coordination between different handlers
 	// You may NOT add a handler *after* the SharedInformer is running.  That will result in an error being returned.
 	// TODO we should try to remove this restriction eventually.
 	AddEventHandler(handler ResourceEventHandler) error
-	GetStore() cache.Store
+	GetIndexer() cache.Indexer
 	// GetController gives back a synthetic interface that "votes" to start the informer
 	GetController() ControllerInterface
 	Run(stopCh <-chan struct{})
 	HasSynced() bool
 }
 
-type SharedIndexInformer interface {
-	SharedInformer
-
-	AddIndexer(indexer cache.Indexer) error
```

Comment:
I think there are two options:
- either what you are suggesting
- or change this method to AddIndexers(indexers cache.Indexers)

The latter is probably preferred, for consistency (we are always passing Indexers in all other places).